File: server/types/dataTypes.go
package types

type SupplierResult struct {
	PresignedLogoUrl      string `json:"presignedLogoUrl"`
	LogoUrl               string `json:"logoUrl"`
	PresignedMarketingUrl string `json:"presignedMarketingUrl"`
	MarketingUrl          string `json:"marketingUrl"`
}

type ModelResult struct {
	PresignedUrl string `json:"presignedUrl"`
	ImageUrl     string `json:"imageUrl"`
}

type PartsModelResult struct {
	PresignedImageUrl string `json:"presignedUrl"`
	ImageUrl          string `json:"imageUrl"`
	PresignedLinkUrl  string `json:"linkUrl"`
}

type EmailData struct {
	Name    string `json:"name"`
	Email   string `json:"email"`
	Message string `json:"message"`
}

type UserData struct {
	Email    string `json:"email"`
	Password string `json:"password"`
	Role     string `json:"role"`
}

type LoginAuth struct {
	Username string `json:"username"`
	Password string `json:"password"`
}

type CustomClaims struct {
	Admin bool `json:"admin"`
}

type User struct {
	Username     string       `json:"username"`
	Password     string       `json:"password"`
	Email        string       `json:"email"`
	RawId        string       `json:"rawId"`
	ProviderId   string       `json:"ProviderId"`
	CustomClaims CustomClaims `json:"customClaim"`
}

File: client/src/types/dataTypes.ts
import { Employee, Privacy, Terms, Timeline } from './aboutTypes';
import { Blog, Exhibition } from './blogTypes';
import { Carousel, LineItem, MachineRegistration, WarrantyClaim, WarrantyParts } from './miscTypes';
import { Machine, Product, Sparepart, Supplier } from './supplierTypes';
import { Video, VideoWebUrl } from './videoTypes';

export interface SupplierResult {
    presignedLogoUrl: string;
    logoUrl: string;
    presignedMarketingUrl: string;
    marketingUrl: string;
}

export interface ModelResult {
    presignedUrl: string;
    imageUrl: string;
}

export interface EmailData {
    name: string;
    email: string;
    message: string;
}

export interface UserData {
    email: string;
    password: string;
    role: string;
}

export interface CustomClaims {
    admin: boolean;
}

export interface User {
    email: string;
    rawId: string;
    providerId: string;
    CustomClaims: CustomClaims;
}

export interface ResourceData {
    id: string;
    route: string;
    queryKey: string;
}

export interface ResourceEntry<T> {
    endpoint: string;
    queryKey: string;
    type?: T;
}

export interface Resources {
    suppliers: ResourceEntry<Supplier>;
    spareparts: ResourceEntry<Sparepart>;
    machines: ResourceEntry<Machine>;
    products: ResourceEntry<Product>;
    videos: ResourceEntry<Video | VideoWebUrl>;
    blogs: ResourceEntry<Blog>;
    exhibitions: ResourceEntry<Exhibition>;
    employees: ResourceEntry<Employee>;
    timelines: ResourceEntry<Timeline>;
    terms: ResourceEntry<Terms>;
    privacys: ResourceEntry<Privacy>;
    lineitems: ResourceEntry<LineItem>;
    carousels: ResourceEntry<Carousel>;
    registrations: ResourceEntry<MachineRegistration>;
    warranty: ResourceEntry<WarrantyClaim | WarrantyParts>;
    supplierMachine: ResourceEntry<Machine>;
    users: ResourceEntry<User | UserData>;
}

export interface FormField {
    name: string;
    label: string;
    type: string;
    options?: { label: string; value: string | undefined }[];
    placeholder: string;
    defaultValue?: string;
}
